File: nas_spec/code_samples/Java/marketplace@entities/post.java
// For more information please refer to https://github.com/checkout/checkout-sdk-java
import com.checkout.CheckoutApiException;
import com.checkout.CheckoutArgumentException;
import com.checkout.CheckoutAuthorizationException;
import com.checkout.CheckoutSdk;
import com.checkout.Environment;
import com.checkout.common.Address;
import com.checkout.common.CountryCode;
import com.checkout.four.CheckoutApi;
import com.checkout.marketplace.ContactDetails;
import com.checkout.marketplace.DateOfBirth;
import com.checkout.marketplace.Document;
import com.checkout.marketplace.DocumentType;
import com.checkout.marketplace.Identification;
import com.checkout.marketplace.Individual;
import com.checkout.marketplace.OnboardEntityRequest;
import com.checkout.marketplace.OnboardEntityResponse;
import com.checkout.marketplace.Phone;
import com.checkout.marketplace.Profile;

CheckoutApi api = CheckoutSdk.fourSdk()
    .oAuth()
    .clientCredentials("client_id", "client_secret")
    .environment(Environment.SANDBOX) // or Environment.PRODUCTION
    .scopes(FourOAuthScope.MARKETPLACE) // more scopes available
    .build();

OnboardEntityRequest onboardEntityRequest = OnboardEntityRequest.builder()
    .reference("reference")
    .contactDetails(ContactDetails.builder()
        .phone(Phone.builder().number("2345678910").build())
        .build())
    .profile(Profile.builder()
        .urls(Arrays.asList("https://docs.checkout.com/1", "https://docs.checkout.com/2"))
        .mccs(Collections.singletonList("0742"))
        .build())
    .individual(Individual.builder()
        .firstName("FirstName")
        .lastName("LastName")
        .tradingName("TradingName")
        .registeredAddress(Address.builder()
            .addressLine1("Checkout.com")
            .addressLine1("90 Tottenham Court Road")
            .city("London")
            .state("London")
            .zip("WIT 4TJ")
            .country(CountryCode.GB)
            .build())
        .nationalTaxId("TAX123456")
        .dateOfBirth(DateOfBirth.builder()
            .day(5)
            .month(6)
            .year(1990)
            .build())
        .identification(Identification.builder()
            .nationalIdNumber("AB123456C")
            .document(Document.builder()
                    .back("number")
                    .front("number")
                    .type(DocumentType.DRIVING_LICENSE)
                    .build())
            .build())
        .build())
    .build();

try {
    OnboardEntityResponse response = api.marketplaceClient().createEntity(onboardEntityRequest).get();
} catch (CheckoutApiException e) {
    // API error
    String requestId = e.getRequestId();
    int statusCode = e.getHttpStatusCode();
    Map<String, Object> errorDetails = e.getErrorDetails();
} catch (CheckoutArgumentException e) {
    // Bad arguments
} catch (CheckoutAuthorizationException e) {
    // Invalid authorization
}

File: nas_spec/code_samples/PHP/payment-links/post.php
<?php
//For more information please refer to https://github.com/checkout/checkout-sdk-php

use Checkout\\CheckoutApiException;
use Checkout\\CheckoutAuthorizationException;
use Checkout\\CheckoutSdk;
use Checkout\\Common\\Address;
use Checkout\\Common\\Country;
use Checkout\\Common\\Currency;
use Checkout\\Common\\CustomerRequest;
use Checkout\\Common\\Phone;
use Checkout\\Common\\Product;
use Checkout\\Environment;
use Checkout\\Payments\\BillingInformation;
use Checkout\\Payments\\Links\\PaymentLinkRequest;
use Checkout\\Payments\\PaymentRecipient;
use Checkout\\Payments\\ProcessingSettings;
use Checkout\\Payments\\RiskRequest;
use Checkout\\Payments\\ShippingDetails;
use Checkout\\Payments\\ThreeDsRequest;

$api = CheckoutSdk::builder()->staticKeys()
        ->environment(Environment::sandbox())
        ->secretKey("secret_key")
        ->build();

$phone = new Phone();
$phone->country_code = "+1";
$phone->number = "415 555 2671";

$address = new Address();
$address->address_line1 = "CheckoutSdk.com";
$address->address_line2 = "90 Tottenham Court Road";
$address->city = "London";
$address->state = "London";
$address->zip = "W1T 4TJ";
$address->country = Country::$GB;

$customerRequest = new CustomerRequest();
$customerRequest->email = "";
$customerRequest->name = "Name";

$billingInformation = new BillingInformation();
$billingInformation->address = $address;
$billingInformation->phone = $phone;

$shippingDetails = new ShippingDetails();
$shippingDetails->address = $address;
$shippingDetails->phone = $phone;

$recipient = new PaymentRecipient();
$recipient->account_number = "1234567";
$recipient->dob = "1985-05-15";
$recipient->last_name = "Last";
$recipient->zip = "12345";

$product = new Product();
$product->name = "Product";
$product->quantity = 1;
$product->price = 10;

$products = array($product);

$theeDsRequest = new ThreeDsRequest();
$theeDsRequest->enabled = false;
$theeDsRequest->attempt_n3d = false;

$processing = new ProcessingSettings();
$processing->aft = true;

$risk = new RiskRequest();
$risk->enabled = false;

$request = new PaymentLinkRequest();
$request->amount = 1000;
$request->reference = "reference";
$request->currency = Currency::$GBP;
$request->description = "Description";
$request->customer = $customerRequest;
$request->shipping = $shippingDetails;
$request->billing = $billingInformation;
$request->recipient = $recipient;
$request->processing = $processing;
$request->products = $products;
$request->risk = $risk;
$request->locale = "en-GB";
$request->three_ds = $theeDsRequest;
$request->capture = true;

try {
    $response = $api->getPaymentLinksClient()->createPaymentLink($request);
} catch (CheckoutApiException $e) {
    // API error
    $error_details = $e->error_details;
    $http_status_code = isset($e->http_metadata) ? $e->http_metadata->getStatusCode() : null;
} catch (CheckoutAuthorizationException $e) {
    // Bad Invalid authorization
}
